src/file_system/error.rs

```diff
@@ -66,12 +66,12 @@ pub enum PathError {
 pub enum LabelError {
     /// An error signifying that a [`Label`](super::path::Label) is contains
     /// invalid UTF-8.
-    #[error("Label contains invalid UTF-8: {0}")]
+    #[error("label contains invalid UTF-8: {0}")]
     InvalidUTF8(String),
     /// An error signifying that a [`Label`](super::path::Label) contains a `/`.
-    #[error("Label contains a slash: {0}")]
+    #[error("label contains a slash: {0}")]
     ContainsSlash(String),
```

Comment:
`anyhow` is much better than `String`, it's a valid `std::error::Error`, so it can be returned as a `source` and it's more elastic (it can contain any source, not only a `String`)

Same for `InvalidUTF8`.

Reply:
Hmmm so the point is that it's the underlying `String` for the `Label`, which isn't a valid label. So it's not so much a message but the data. Would I still use `anyhow::Error`? I feel like I wouldn't :thinking:

Reply:
Ah, I haven't noticed that it's supposed to contain the invalid label value! I'm afraid that I may not be the only one who will get confused reading this code. How about we make it something like this?
```suggestion
    ContainsSlash {
        label_value: String
    },
```

Reply:
Great call :)